src/plotting/plotSettings/plotSettingsController.component.test.tsx

```diff
@@ -117,17 +68,91 @@ describe('Plot Settings component', () => {
     jest.clearAllMocks();
   });
 
-  it('renders plot settings form correctly (x-axis tab selected)', () => {
-    const view = createView();
-
-    expect(view.asFragment()).toMatchSnapshot();
-  });
-
-  it('renders plot settings form correctly (y-axis tab selected)', async () => {
-    const view = createView();
-    await user.click(screen.getByRole('tab', { name: 'Y' }));
-
-    expect(view.asFragment()).toMatchSnapshot();
+  describe('snapshots', () => {
+    beforeAll(() => {
+      jest.resetModules();
+      jest.doMock('./plotTitleField.component', () => (props) => (
```

Comment:
Just curious - where are the `mock-*` components coming from? Is it a specific `jest.doMock` syntax that I am unaware of?

Reply:
The mock components are so that we can snapshot test the plotSettingsController without snapshotting all the underlying components. The tests therefore check that the correct props are being passed to the respective components. We're using `doMock` instead of `mock` so that the mocked components are only mocked in the snapshot tests, and in the other tests we're using the real components (and so can interact with them). The `mock-*` components are just nonsense JSX essentially.